sql/hive/src/main/scala/org/apache/spark/sql/hive/MetastoreRelation.scala

```diff
@@ -141,7 +142,16 @@ private[hive] case class MetastoreRelation(
           sparkSession.sessionState.conf.defaultSizeInBytes
         })
     }
-  )
+    val tableParams = hiveQlTable.getParameters
+    val rowCount = tableParams.get(AnalyzeTableCommand.ROW_COUNT)
+    if (rowCount != null && rowCount.toLong >=0) {
```

Comment:
actually, I will need the if block to decide whether to compute total size when we have column stats. 
Although we can still use Option(rowCount).map(some function)... for the second property estimatedSize, this looks not intuitive. The if-else is easier to read.
So I'll keep the if-else block, what do you think?